### ISSUE TITLE
[Bug]: UI Mode behave incorrectly on macOS

### ISSUE THREAD

**Issue by lucydodo** [sqlitebrowser/sqlitebrowser#3570]
### What did you do?

We've seen some issues with 'Dark Mode' recently, so I tested for all cases of UI settings in macOS and in our app.

### What did you expect to see?

Consistent UI experience

### What did you see instead?

| App UI Setting | macOS UI Setting | Screenshot                                                                                                                                                           |
| -------------- | ---------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Auto           | Light            | <img width="1149" alt="Screenshot 2024-02-20 at 19 24 27" src="https://github.com/sqlitebrowser/sqlitebrowser/assets/64626662/b1b2513f-0d62-46b0-a5c1-52115ee3667f"> |
| Light          | Light            | <img width="1149" alt="Screenshot 2024-02-20 at 19 24 33" src="https://github.com/sqlitebrowser/sqlitebrowser/assets/64626662/9a9e6fb3-a119-414b-83ca-eba1c4b01ee2"> |
| Dark           | Light            | <img width="1149" alt="Screenshot 2024-02-20 at 19 24 39" src="https://github.com/sqlitebrowser/sqlitebrowser/assets/64626662/4e1dd8b9-e9fb-447c-8a27-2d4105c2e223"> |
| Auto           | Dark             | <img width="1149" alt="Screenshot 2024-02-20 at 19 25 02" src="https://github.com/sqlitebrowser/sqlitebrowser/assets/64626662/55d409eb-f79f-484d-b5f0-852ccb9c5cd0"> |
| Light          | Dark             | <img width="1149" alt="Screenshot 2024-02-20 at 19 25 08" src="https://github.com/sqlitebrowser/sqlitebrowser/assets/64626662/25e1cb60-4c70-4eda-ba02-7d52b53d9b23"> |
| Dark           | Dark             | <img width="1149" alt="Screenshot 2024-02-20 at 19 25 12" src="https://github.com/sqlitebrowser/sqlitebrowser/assets/64626662/ccc7b1b6-1a19-4e82-9c5f-88dbb3fdf558"> |

I think (but haven't checked the code flow yet) that if user set it to 'Follow the desktop UI', the UI changes are made by the Qt framework, and if user select the 'Light' or 'Dark' option, the changes are made by the other stylesheet.

### DB4S Version

3.13.99 (nightly)

### What OS are you seeing the problem on?

MacOS

### OS version

macOS 14.3.1 (23D60)

### Prevention against duplicate issues

- [X] I have searched for similar issues

**Comment by justinclift** [comment]
Oh that looks weird.  We have two different dark modes.

Wonder if that's a "feature" rather than a bug? :wink:

(not really, it's definitely a bug)

**Comment by lucydodo** [comment]
Yes, it's not what we intended anyway, so I guess it's a 'bug'. I'll try to check how it works on other operating systems first.

**Comment by rafrafek** [comment]
In my case there is a glitch with an element displaying in light theme:

![image](https://github.com/sqlitebrowser/sqlitebrowser/assets/23004737/bc3021db-b1ff-4cc6-bf8f-560a6e31ecf1)

**Comment by justinclift** [comment]
@rafrafek Does that `S...` problem still show up if you resize the window to be wider?

I'm still thinking it's a bug though.  Am just wondering. :smile:

**Comment by rafrafek** [comment]
Yep, still there after resizing

<img width="1722" alt="image" src="https://github.com/sqlitebrowser/sqlitebrowser/assets/23004737/3785eb57-937d-4273-a3f3-5ddfc9a00fb2">

**Comment by justinclift** [comment]
k.  Sounds like we might need to add some code that manually changes the width of the tab titles to be longer. :smile:

**Comment by rafrafek** [comment]
I forgot to mention I'm on 3.12.2, so the behavior may be different on the master branch. I'll try to test the master branch tomorrow.

**Comment by rafrafek** [comment]
Same thing on both 3.12.2 and on the latest build:

<img width="1107" alt="image" src="https://github.com/sqlitebrowser/sqlitebrowser/assets/23004737/2778b4f3-2222-46e4-a66c-b59b61751adb">

**Comment by mgrojo** [comment]
The screenshots in @lucydodo's description look good to me. It's working as designed :)
Glitchs reported by @rafrafek are another matter. They should be fixed.
